zio-kafka/src/main/scala/zio/kafka/consumer/ConsumerSettings.scala

```diff
@@ -20,22 +21,17 @@ import zio.kafka.security.KafkaCredentialStore
  *   be much larger than the pollTimeout and the time it takes to process chunks of records. If your consumer is not
  *   subscribed for long periods during its lifetime, this timeout should take that into account as well. When the
  *   timeout expires, the plainStream/partitionedStream/etc will fail with a [[Consumer.RunloopTimeout]].
- * @param enableOptimisticResume
- *   When `true` (the default) zio-kafka predicts whether a stream needs more data, slightly ahead of time. Zio-kafka
- *   pauses partitions for which the associated stream is processing previously fetched data. When the stream needs more
- *   data, the partition is resumed. With this feature enabled, partitions are also resumed when it is likely that the
- *   stream needs more data in the next poll.
  */
 final case class ConsumerSettings(
   bootstrapServers: List[String],
   properties: Map[String, AnyRef] = Map.empty,
   closeTimeout: Duration = 30.seconds,
   pollTimeout: Duration = 50.millis,
+  perPartitionChunkPrefetch: Int = 2,
```

Comment:
Do we really want to put this back?

Reply:
My plan is to remove it in 803 again, this is just a revert of 809 (as clean as possible)

Reply:
I'll leave the approval of this PR to Erik at it's his work if that's ok for you

Reply:
If we go back to this, we should also bring back this setting.
However, I still think we can do better. E.g. by making this maximum prefetch (like in fs2-kafka):

```suggestion
  perPartitionChunkMaxPrefetch: Int = 2,
```

Reply:
We'll leave it for another PR.